### DISCUSSION TITLE
parentId of OPT-300-Procedure-SProvider

### DISCUSSION THREAD

**Discussion by JasperBeaurain** [OP-TED/eForms-SDK#132] [Q&A] [answered]
During conversion from our visual model to our conceptual model, we are experiencing issues with the field "OPT-300-Procedure-SProvider".
Could it be that the parentNodeId for this field should be "ND-ServiceProviderParty" instead of "ND-ServiceProvider" (and that the relative XPath should hence start with cac:Party instead of cac:ServiceProviderParty)?

**Comment by rousso** [comment, marked as answer]
Hi Jasper,

Thank you for the [excellent explanation](https://github.com/OP-TED/eForms-SDK/discussions/126#discussioncomment-3812838) you provided on this topic in #126! 

I understand your problem. Indeed when "reusing" the created nodes in the conceptual model we need to be a bit "smart" about it. Reusing either all or none of the existing nodes is not going to work. 

My first thought on this is that you need to use instance identifiers. What I mean is that a `display-group` which is not repeatable should not be a problem to map to a `node` in the conceptual model because it appears only once. However, when you map a repeatable `display-group` to a `node` you need to tag that node with an instance identifier of the repeatable `display-group` so that you can find that `node` again when you need to reuse it.

I need to reflect on this for a while with my team to **double-check** first of all that the hierarchy of the specific `display-groups` `fields` and `nodes` is provided correctly in the SDK but to also analyse a bit the situation so that I can give you a meaningful answer.

I will answer to you in #126 where your clear description of the problem will also help others follow this important point more clearly.

Thank you. I really appreciate your feedback on this.

Regards,
Yannis

**Reply by JasperBeaurain** [reply]
Hi Yannis,

Thanks for your response!

Indeed, keeping track of which repeated node you need to place children under is necessary. We inplemented this by keeping indices of the (repeatable) groups (that map to repeatable nodes). That way, we know how to place a child node under its correct parent node. For most repeated nodes, this solves the problem.

The issue with ServiceProviderParty & BT-300-Procedure-SProvider is a little bit more complex however: under ND-ServiceProvider there are two elements (ND-ServiceProvderParty and BT-300-Procedure-SProvider) that both correspond to xml element  cac:ServiceProviderParty, but that info is missing in the conceptual model so keeping indices does not help.

I think that if BT-300-Procedure-Sprovider would be a child of ND-ServiceProvderParty in fields.json, the indices approach would work correctly. So maybe this is indeed a mistake in the metadata?

It's not very easy to explain this issue, but I hope you'll understand :smile:. If you need any further details, I'm available & you can contact me here.

Thanks for your input & I hope to hear from you soon. Sorry for posting about this in 2 different places :wink:.

Thanks!
Jasper

**Reply by JasperBeaurain** [reply]
For anyone reading this: the issue was discussed further in #126 
OP-TED found the cause of the issue, as was explained here: https://github.com/OP-TED/eForms-SDK/discussions/126#discussioncomment-3828331
